controllers/songs.js

```diff
@@ -17,10 +28,39 @@ const createSong = async (req, res) => {
 };
 
 const updateSong = async (req, res) => {
-  res.send('update a song');
+  const { userId } = req.user;
+  const { id: songId } = req.params;
+  const { title, artist } = req.body;
+  if (title === '' || artist === '') {
+    throw new BadRequestError('Title or artist cannot be empty');
```

Comment:
```suggestion
    throw new BadRequestError('Title or artist must be provided');
```

title or artist can be empty! But they can't _both_ be empty!